RecoTracker/LSTCore/src/alpaka/Event.dev.cc

```diff
@@ -94,10 +93,35 @@ void Event::addHitToEvent(std::vector<float> const& x,
     hitsInGPU_->setData(*hitsBuffers_);
   }
 
-  if (!rangesInGPU_) {
-    rangesInGPU_.emplace();
-    rangesBuffers_.emplace(nModules_, nLowerModules_, devAcc_, queue_);
-    rangesInGPU_->setData(*rangesBuffers_);
+  if (!rangesDC_) {
+    std::array<int, 2> const ranges_sizes{{static_cast<int>(nModules_), static_cast<int>(nLowerModules_ + 1)}};
+    rangesDC_.emplace(ranges_sizes, queue_);
+
+    auto ranges = rangesDC_->view<ObjectRangesSoA>();
```

Comment:
do I understand correctly that if we can get a view the size of `ObjectRangesSoA` we can memset it to zero directly?

Reply:
Yeah, I wonder if there's an easy way to do that. There are so many memsets that it might even be faster to set the entire buffer (which also includes the other SoA)

Reply:
the other part is probably around x4 smaller (modules vs lower modules and also `[2]` )
perhaps it's simpler to just initialize the full buffer